backend/src/apiserver/storage/experiment_store.go

```diff
@@ -26,33 +26,69 @@ type ExperimentStore struct {
 	resourceReferenceStore *ResourceReferenceStore
 }
 
-func (s *ExperimentStore) ListExperiments(opts *list.Options) ([]*model.Experiment, string, error) {
-	errorF := func(err error) ([]*model.Experiment, string, error) {
-		return nil, "", util.NewInternalServerError(err, "Failed to list experiments: %v", err)
+// Runs two SQL queries in a transaction to return a list of matching experiments, as well as their
+// total_size. The total_size does not reflect the page size.
+func (s *ExperimentStore) ListExperiments(opts *list.Options) ([]*model.Experiment, int32, string, error) {
+	errorF := func(err error) ([]*model.Experiment, int32, string, error) {
+		return nil, 0, "", util.NewInternalServerError(err, "Failed to list experiments: %v", err)
 	}
 
-	sql, args, err := opts.AddToSelect(sq.Select("*").From("experiments")).ToSql()
+	sqlBuilder := opts.AddFilterToSelect(sq.Select("*").From("experiments"))
+
+	// SQL for getting total size
+	countSql, countArgs, err := sq.Select("count(*)").FromSelect(sqlBuilder, "rows").ToSql()
```

Comment:
@IronPan 

I agree with Yang here. Could we simplify the query so that we avoid a full table scan by the subquery?